src/ILCompiler.Compiler/src/Compiler/MultiFileCompilationModuleGroup.cs

```diff
@@ -181,5 +210,32 @@ private HashSet<EcmaModule> InputModules
                 return _compilationModuleSet;
             }
         }
+
+        /// <summary>
+        /// Validates that it will be possible to generate '<paramref name="method"/>' based on the types 
+        /// in its signature. Unresolvable types in a method's signature prevent RyuJIT from generating
+        /// even a stubbed out throwing implementation.
+        /// </summary>
+        private static void CheckCanGenerateMethod(MethodDesc method)
+        {
+            MethodSignature signature = method.Signature;
+
+            CheckTypeCanBeUsedInSignature(signature.ReturnType);
+
+            for (int i = 0; i < signature.Length; i++)
+            {
+                CheckTypeCanBeUsedInSignature(signature[i]);
+            }
+        }
+
+        private static void CheckTypeCanBeUsedInSignature(TypeDesc type)
+        {
+            MetadataType defType = type as MetadataType;
+
+            if (defType != null)
+            {
+                var unused = defType.ContainsGCPointers;
```

Comment:
Does this cause any build warnings?

Maybe exposing `ComputeTypeContainsGCPointers` (for parity with the methods that compute layout) wouldn't be that bad.